root/root.go

```diff
@@ -71,6 +85,28 @@ func (root *Root) handlePortUpdate(input interface{}, ctx common.RestContext) (i
 	return nil, nil
 }
 
+// Handler for the /auth URL
+func (root *Root) handleAuth(input interface{}, ctx common.RestContext) (interface{}, error) {
+	cred := input.(*common.Credential)
+	// We assume just username/password for now
+	roles, err := root.store.Authenticate(cred)
+	if err != nil {
+		return nil, err
+	}
+	token := jwt.New(jwt.SigningMethodHS256)
+	rolesStr := make([]string, len(roles))
+	for i in range roles {
+		rolesStr[i] = roles[i].Name
+	}
+	token.Claims["roles"] = rolesStr
+	token.Claims["iat"] = time.Now().Unix()
+	// TODO make this configurable?
+	token.Claims["exp"] = time.Now().Add(time.Second * 3600 * 24).Unix()
```

Comment:
We should make this configurable of course. time.Second is a "Duration"... It's a go way for time arithmetic, kinda like python's arrow and java joda?